implementation/src/main/java/io/smallrye/health/InetAddressHealthCheck.java

```diff
@@ -0,0 +1,67 @@
+package io.smallrye.health;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+
+/**
+ * Health check implementation to check if host is reachable using {@code java.net.InetAddress.isReachable} method.
+ *
+ * <pre>
+ * {@code
+ * @Produces
+ * @ApplicationScoped
+ * @Liveness
+ * HealthCheck check1() {
+ *   return new InetAddressHealthCheck("service.com");
+ * }
+ * }
+ * </pre>
+ *
+ * @see InetAddress
+ */
+public class InetAddressHealthCheck implements HealthCheck {
+
+    private static final String DEFAULT_NAME = "Internet Address Check";
+    private static final int DEFAULT_TIMEOUT = 2000;
+
+    private String host;
+    private String name;
+    private int timeout;
+
+    public InetAddressHealthCheck(String host) {
+        this(host, DEFAULT_NAME);
+    }
+
+    public InetAddressHealthCheck(String host, String name) {
+        this.host = host;
+        this.name = name;
+        this.timeout = DEFAULT_TIMEOUT;
+    }
+
+    @Override
+    public HealthCheckResponse call() {
+        final HealthCheckResponseBuilder healthCheckResponseBuilder = HealthCheckResponse.named(this.name);
+        healthCheckResponseBuilder.withData("host", this.host);
+        try {
+            InetAddress addr = InetAddress.getByName(this.host);
+            final boolean reachable = addr.isReachable(this.timeout);
```

Comment:
I'm not familiar with `isReachable`. Can it return false and not throw an exception?

Just wondering under what situations the catch would be triggered?

Reply:
According to javadoc, the exception occurs when there is a timeout or a network exception meanwhile the method might return true or false depending on the reachable of the host. https://docs.oracle.com/javase/8/docs/api/java/net/InetAddress.html#isReachable-int-

Reply:
Ok thanks